policybot/pkg/blobstorage/gcs/store.go

```diff
@@ -28,21 +28,42 @@ type store struct {
 	client *storage.Client
 }
 
-func NewStore(ctx context.Context, gcpCreds []byte) (blobstorage.Store, error) {
-	client, err := storage.NewClient(ctx, option.WithCredentialsJSON(gcpCreds))
-	if err != nil {
-		return nil, fmt.Errorf("unable to create GCS client: %v", err)
-	}
-
+func NewStore(client *storage.Client) blobstorage.Store {
```

Comment:
Can we encode this API in such a way that the GCS stuff is entirely encapsulated? Consider how the Spanner stuff is entirely in the spanner/store.go code so anything above only sees a generic New signature, independent of Spanner. I was hoping for the same layering for the blob store.

Reply:
Done. I had originally done this because I wanted to potentially be able to use a fake/local Spanner with the interface, but that can be addressed later.